python/cudf_polars/cudf_polars/containers/dataframe.py

```diff
@@ -41,7 +41,7 @@ def __init__(self, columns: Sequence[NamedColumn]) -> None:
 
     def copy(self) -> Self:
         """Return a shallow copy of self."""
-        return type(self)(self.columns)
+        return type(self)([c.copy() for c in self.columns])
```

Comment:
This was a bug, we should have been copying the columns because the metadata is carried on the columns.